### DISCUSSION TITLE
Authentication - Restricting users

### DISCUSSION THREAD

**Discussion by dazinator** [polymind-inc/acmebot#441] [Q&A] [answered]
After a fresh install following the getting started guide (thank you) - I have found that I can authenticate with my Azure AD account as expected and see the dashboard.

However, so can another developer.

I'd like to be explicit about which users can login and see this dashboard.
I checked the "app registration" in Azure that was created and there are no "roles" defined for this function app:

![image](https://user-images.githubusercontent.com/3176632/148570830-cd5039b8-f791-4b3e-bf95-7d101f9c98ef.png)

I remember seeing on the ARM template something about a role ID being required (which I entered) but I have yet to see where this role applies so this is slightly confusing.

Is it possible to control which users are allowed to log in to the app? The danger is I don't want a non developer (who happens to be in our Azure AD directory) accessing the dashboard and then revoking certs :-)

**Comment by shibayan** [comment, marked as answer]
The App Role must be created manually.
https://github.com/shibayan/keyvault-acmebot/wiki/App-Role-based-authorization
